File: Leetcode/49-group-anagrams/49-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        vector<vector<string>> ret;
        unordered_map<string, vector<string>> map;

        for(const auto& s: strs){
            string t = s;
            sort(t.begin(), t.end());
            map[t].push_back(s);
        }

        ret.reserve(map.size());
        for(auto& pair: map){
            ret.push_back(std::move(pair.second));
        }
        return ret;
    }
};

File: Leetcode/49-group-anagrams/49-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        if len(strs) == 0: return [[]]
        hash: Dict[Tuple, List[str]] = {}
        for s in strs:
            key = tuple(sorted(s))
            hash[key] = hash.get(key, []) + [s]
        return hash.values()